### PR TITLE
replaces reputation locks with instead being prohibitively expensive based on rep

### PR DESCRIPTION
**Pull request by mc-oofert**
## About The Pull Request

makes reputation locked gear instead be very very expensive and get cheaper as you get to that rep
price initially starts at 3x 0 rep and climbs to 1x as you approach the needed reputation

https://github.com/user-attachments/assets/7696e776-6058-474a-b028-f120c90df45e

## Why It's Good For The Game

Reputation locking you out until youre exactly or above that rep kinda sucks as it locks you out of doing the things you want because you dont have the rep yet so you gotta AFK in a locker or stare into a wall or play the really poorly made objectives

with this change you can sacrifice much more TC to have it early
still prevents roundstart medbay bombing: syndbomb costs 33tc roundstart or so

## Changelog
:cl:
balance: makes reputation locked gear instead be very very expensive and get cheaper as you get to that rep
/:cl:

### CODE REVIEW

**Comment by TheVekter** [comment]
I don't know that I like this, because it feels like an attempt to just quiet remove the side objectives because you don't like them. I feel like it would be more productive to just add more interesting side objectives.

**Comment by castawaynont** [comment]
> I don't know that I like this, because it feels like an attempt to just quiet remove the side objectives because you don't like them. I feel like it would be more productive to just add more interesting side objectives.

the side objectives are a lost cause - you could add a hundred objectives but due to the constraints of being railroaded in what's considered "completion" of them and the amount of rounds played in a day, you can't do anything truly interesting, and they'd get just as stale - not even factoring in the fact that people are just gonna choose the easiest ones above all else (hello merking pets simulatior 13)

**Comment by necromanceranne** [comment]
> I don't know that I like this, because it feels like an attempt to just quiet remove the side objectives because you don't like them. I feel like it would be more productive to just add more interesting side objectives.

Thats not what is taking place. Its basically exchanging one timelock method for another, offsetting rep gain for tc deficits. If anything, I think this system is worse if only because a player might accidentally buy something at markup without realizing, since there isn't a clear indicator what the true price is in the UI.

**Comment by DaydreamIQ** [comment]
There's also the issue that midround/latejoin tots aren't progtot anymore but still get the reputation system for...some reason??? So while its nice to be able to buy a desword earlier its still not really addressing some of the issues with the system

**Comment by mc-oofert** [comment]
not that we can address any of the issues because watermelon would've closed the PRs regardless

**Comment by SimplySmiley123** [comment]
> I don't know that I like this, because it feels like an attempt to just quiet remove the side objectives because you don't like them. I feel like it would be more productive to just add more interesting side objectives.

Like what?
What side objectives could you add at this point that would be interesting in any way? And most importantly worth the effort put into completing the side objective.
There is or rather was (haven't seen it anymore) the kidnap a crew member side objective which paid too little for how randomly hard they could be for anyone to bother. Everyone would do the easy pet killing objectives instead.

**Comment by ElGitificador** [comment]
> I don't know that I like this, because it feels like an attempt to just quiet remove the side objectives because you don't like them. I feel like it would be more productive to just add more interesting side objectives.

most of the rounds end up with almost none to none traitors doing ANY side objectives nowadays, i dont know how many times progtot has to be proven to be a failure for it to be removed/remade

**Comment by moocowswag** [comment]
The worst part of reputation is the soft time lock, you have diminishing returns on reputation gain from completing objectives , this was implemented to stop tots from speedrunning their final objective (oh but heretic can still speedrun their ascension???) and ive witnessed someone actually get punished by admins because they recalled shuttle in order to pursue their final objective which they still couldnt do due to reputation.

File: tgui/packages/tgui/interfaces/Uplink/index.tsx [hunk 1]
```diff
@@ -213,8 +214,17 @@ export class Uplink extends Component<{}, UplinkState> {
...
-        progression_points >= item.progression_minimum;
+      // please sync with real_cost on datum/uplink_item or whatever it was called
+      const real_cost = () => {
+        if (item.progression_minimum <= 0 || !has_progression) return item.cost;
+        var percentage = clamp(
+          progression_points / item.progression_minimum,
+          0,
+          1,
+        );
+        var mult = (1 - percentage) * 3 + percentage * 1;
```
[the remark this answers is not in the document]
Comment (mc-oofert) [reply tgui/packages/tgui/interfaces/Uplink/index.tsx]: no idea how to do that Uplink is extremely convoluted

[the remark this answers is not in the document]
Comment (mc-oofert) [reply tgui/packages/tgui/interfaces/Uplink/index.tsx]: nvm

**Comment by Iamgoofball** [comment]
The entire point of the reputation system is to stop you from roundstart buying a 5/10/20 and throwing it into medical.

**Comment by mc-oofert** [comment]
> The entire point of the reputation system is to stop you from roundstart buying a 5/10/20 and throwing it into medical. Making it not a lock negates this.

still stops it if you ever gave some thought before posting

**Comment by Blacklist897** [comment]
MERGED WITHOUT LOOKING BABY

**Comment by mc-oofert** [comment]
> MERGED WITHOUT LOOKING BABY

my favorite kind of merge

**Pull request merged**